r2dbc-mysql/src/main/java/JasyncClientConnection.kt

```diff
@@ -0,0 +1,66 @@
+package com.github.jasync.r2dbc.mysql
+
+
+import io.r2dbc.spi.Batch
+import io.r2dbc.spi.Connection
+import io.r2dbc.spi.IsolationLevel
+import io.r2dbc.spi.Statement
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Flowable
+import org.reactivestreams.Publisher
+import java.util.function.Supplier
+import com.github.jasync.sql.db.Connection as JasyncConnection
+
+class JasyncClientConnection(private val jasyncConnection: JasyncConnection) : Connection, Supplier<JasyncConnection> {
```

Comment:
Do you have support for MySQL's PING? Adding this functionality here would be a preparation for r2dbc/r2dbc-spi#54.

Reply:
the connection itself don't have ping, but there is a connection factory that knows how to test the connection: https://github.com/jasync-sql/jasync-sql/blob/dde3f6afeaa325189ed0467bfd1342bdce9adde9/db-async-common/src/main/java/com/github/jasync/sql/db/pool/ConnectionFactory.kt#L79

Reply:
The idea behind is to check for e.g. pooling implementations whether the connection works without actually executing SQL.

Reply:
no, there is no ping support, but there is a 'validate' check that don't do any network action: https://github.com/jasync-sql/jasync-sql/blob/dde3f6afeaa325189ed0467bfd1342bdce9adde9/db-async-common/src/main/java/com/github/jasync/sql/db/pool/ConnectionFactory.kt#L49